src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!(target instanceof Herbivore) || target.hidden) {
+      return; // Can't bite if it's not a herbivore or the target is hiding.
+    }
+
+    target.health -= 50;
+
+    if (target.health <= 0) {
+      Animal.removeDeadAnimals();
```

Comment:
The `removeDeadAnimals` method is called every time a `bite` occurs, even if the target's health is not below zero. Consider moving the call to `removeDeadAnimals` inside the `if` block to ensure it is only called when the target's health is less than or equal to zero.